crates/kitsune_p2p/types/src/tx2/tx2_utils/tx_url.rs

```diff
@@ -10,6 +12,12 @@ impl TxUrl {
     pub fn as_url2(&self) -> &url2::Url2 {
         &self.0
     }
+
+    /// Construct from a string which is known to be a valid URL.
+    /// Panics if the URL is not parseable.
+    pub fn from_str_unsafe(s: &str) -> Self {
+        url2::Url2::parse(s).into()
+    }
```

Comment:
```suggestion
    pub fn from_str_panicking(s: &str) -> Self {
        url2::Url2::parse(s).into()
    }
```

panics != unsafe

Reply:
Isn't it the rule that From will panic if the conversion fails?